### DISCUSSION TITLE
Make "history" permission optional

### DISCUSSION THREAD

**Discussion by cdrini** [proginosko/LeechBlockNG#480] [Ideas]
[Looking through the code](https://github.com/search?q=repo%3Aproginosko%2FLeechBlockNG%20history&type=code), it seems like the `browser.history` API is only used for the "add blocked page to browser history" option. The history permission looks a little scary, and isn't really clear why it's necessary when it's being installed (see below). Since this option is [disabled by default](https://github.com/proginosko/LeechBlockNG/blob/f46a37d2d8fcf0a077c639f4f6d9c1305925f9b4/common.js#L55), would it be possible to make this permission optional?

Proposal:

- In the manifest.json , move "history" from `permissions` to [`optional_permissions`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/optional_permissions)
- Add some JS to options.js to request the optional permission when the user sets the option to true.

This makes it really easy for the user to understand why the permission is needed, and makes them more likely to install the app since it requires fewer permissions!

Current "scary" permission:
![image](https://github.com/user-attachments/assets/43d2c224-3fad-4ebd-ba72-28d3815bb34d)
![image](https://github.com/user-attachments/assets/929b1477-acff-4a24-9593-f3aca779a669)

**Comment by proginosko** [comment]
History permission now optional in version 1.6.9 (uploaded to AMO, awaiting approval). Also downloads permission no longer required.

**Reply by leoheitmannruiz** [reply]
Thanks!
